nodejs/device/src/index.ts

```diff
@@ -359,24 +395,34 @@ class DeviceApp {
         },
       })
     );
+    logger.info(`Sending message with serviceAckId: ${serviceAckId}`);
     this.client
       .sendEvent(message)
       .catch(() => {
-        console.warn(
-          "Error sending message with serviceAckId: " + serviceAckId
+        logger.error(
+          `Error sending message with serviceAckId: ${serviceAckId}`
         );
       })
       .finally(() => {
         this.serviceAckWaitList.add(serviceAckId);
+        if (
+          this.serviceAckWaitList.size > settings.sendMessageAllowedFailureCount
+        ) {
+          return this.stopDeviceApp(
```

Comment:
Do you need the `return` here?  Isn't that deceptive because you're not really returning anything and the code should never reach the point of returning anyway.